include/sparrow/array/array_data_factory.hpp

```diff
@@ -487,21 +489,45 @@ namespace sparrow
         }
     }
 
+
     /**
      * \brief Creates a default array_data object with the specified layout and values.
      *
      * @tparam Layout The layout of the array_data object.
      * @tparam ValueRange The type of the input range for the values.
      * @param values The input range of values for the array_data object.
-     * @param bitmap The bitmap type for the array_data object.
-     * @param offset The offset for the array_data object.
      * @return A new array_data object with the specified layout, values, bitmap, and offset.
      */
     template <arrow_layout Layout, std::ranges::input_range ValueRange>
-        requires(!mpl::constant_range<ValueRange>)
     array_data
-    make_default_array_data(ValueRange&& values, const array_data::bitmap_type& bitmap, std::int64_t offset)
+    make_default_array_data(ValueRange&& values)
+    {
+        const std::size_t size = std::ranges::size(values);
+        array_data::bitmap_type bitmap(size, true);
+        const std::int64_t offset = 0;
+        return make_default_array_data<Layout>(std::forward<ValueRange>(values), std::move(bitmap), offset);
+    }
+
+
+    /**
+     * \brief Creates a default array_data object with the specified layout and values.
+     * 
+     * @tparam Layout The layout of the array_data object.
+     * @tparam T The type of the value to be repeated.
+     * 
+     * @param n The number of times the value should be repeated.
+     * @param value The value to be repeated.
+     */
+    template <arrow_layout Layout, class T>
+    requires is_arrow_base_type_extended<std::decay_t<T>>
+    array_data make_default_array_data( 
+        typename Layout::size_type n
+        ,  T && value)
     {
-        return make_default_array_data<Layout>(std::as_const(values), bitmap, offset);
+        auto repeated_range = std::ranges::iota_view{size_t(0),size_t(n)} | std::views::transform([&](auto) { return value; });
+
+        return make_default_array_data<Layout>(repeated_range);
     }
```

Comment:
All constexpr

Reply:
are you suggesting to make the whole `make_default_array_data` function and everything this function calls constexpr?